wiki/Long identifiers.md

```diff
@@ -0,0 +1,176 @@
+	very_long_name_with_length_greater_than_63_but_less_equal_than_128_that_we_would_like_to_test
+) values(41, 'foo')
+```
+```tsql
+insert into tab1 (
+	id1,
+	very_long_name_with_length_greafcb500b635caaad2e5c26272eafc29e4
+) values(42, 'bar')
+```
+
+Column names in result set
+--------------------------
+
+If either long or short name of a column is specified explicitly in `select` query, then the short name will be returned in result set metadata. Two following queries give the same output:
+
+```tsql
+select 
+    id1,
+    very_long_name_with_length_greater_than_63_but_less_equal_than_128_that_we_would_like_to_test
+from tab1
+```
+```tsql
+select 
+    id1,
+    very_long_name_with_length_greafcb500b635caaad2e5c26272eafc29e4
+from tab1
+```
+```
+id1         very_long_name_with_length_greafcb500b635caaad2e5c26272eafc29e4
+----------- ---------------------------------------------------------------
+41          foo
+42          bar
+```
+
+Though, if `select *` is used, then the truncated "PostgreSQL native" version of the long name is included in result set metadata:
+
+```tsql
+select * from tab1
+```
+```
+id1         very_long_name_with_length_greater_than_63_but_less_equal_than_
+----------- ---------------------------------------------------------------
+41          foo
+42          bar
+```
+
+Converting between long and short names
+---------------------------------------
+
+For most types of idenifiers (with index and constraint names being the exception) system function `babelfish_truncate_identifier` can be used to compute short name from a long one:
+
+```tsql
+select sys.babelfish_truncate_identifier('very_long_name_with_length_greater_than_63_but_less_equal_than_128_that_we_would_like_to_test')
+```
+```
+babelfish_truncate_identifier
+------------------------------------------------------------------------------------------------------------------------------------------------------------
+very_long_name_with_length_greafcb500b635caaad2e5c26272eafc29e4
+```
+
+For column names the long form can be looked up in PostgreSQL system catalogs using the following query. Because column name is stored in PostgreSQL `ARRAY` column we need to use a number of functions to extract it. On PostgreSQL connection this can be simplified with array operators.
+
+```tsql
+select split_part(array_to_string(trim_array(at.attoptions, array_length(at.attoptions, 1) - 1), ''), '=', 2) as original_name
+from pg_catalog.pg_class cl
+join pg_catalog.pg_attribute at
+on cl.oid = at.attrelid
+where cl.relname = 'tab1'
+and at.attname = 'very_long_name_with_length_greafcb500b635caaad2e5c26272eafc29e4'
+```
+```
+original_name
+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+very_long_name_with_length_greater_than_63_but_less_equal_than_128_that_we_would_like_to_test
+```
+
+Index and constraint names
```

Comment:
Thanks for adding this section, I looked at the exact logic for computing the names of index/constraint [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/BABEL_4_X_DEV/contrib/babelfishpg_tsql/src/backend_parser/gram-tsql-epilogue.y.c#L44) and it looks like the formula is like this:
```
sys.babelfish_truncate_identifier (index_name + table_name + md5(index_name))
```
Do you think if we should mention this as well?

Reply:
Yes, I've added this query to the index section - [updated doc](https://github.com/babelfish-for-postgresql/babelfish-for-postgresql/blob/1933de922de3a2c8491e5e12c67d81ac34c0973d/wiki/Long%20identifiers.md).